protobuf/lib/src/protobuf/coded_buffer_reader.dart

```diff
@@ -164,19 +164,22 @@ class CodedBufferReader {
     return _buffer[_bufferPos - 1];
   }
 
-  int _readRawVarint32([bool signed = true]) {
+  int _readRawVarint32(bool signed) {
     // Read up to 10 bytes.
-    int bytes = _currentLimit - _bufferPos;
+    int bufferPos = _bufferPos;
+    int bytes = _currentLimit - bufferPos;
     if (bytes > 10) bytes = 10;
     int result = 0;
     for (int i = 0; i < bytes; i++) {
```

Comment:
If this loop shows up on the profile we could even consider unrolling it for small values of `bytes`

Reply:
Maybe it would help. The issue is that the AOT compiler currently cannot inline the `buffer_[...]` access because it doesn't know which representation it is (_ExternalUint8List, _Uint8ArrayView, _Uint8List, ...)